src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala

```diff
@@ -17,45 +17,54 @@ import scala.util.control.NonFatal
 sealed trait Arg extends (ScalafixArguments => ScalafixArguments)
 
 object Arg {
-  case class ToolClasspath(classLoader: URLClassLoader) extends Arg {
+  sealed trait CacheKey
+
+  case class ToolClasspath(classLoader: URLClassLoader)
+      extends Arg
+      with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withToolClasspath(classLoader)
   }
 
-  case class Rules(rules: Seq[String]) extends Arg {
+  case class Rules(rules: Seq[String]) extends Arg with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withRules(rules.asJava)
   }
 
-  case class Paths(paths: Seq[Path]) extends Arg {
+  case class Paths(paths: Seq[Path]) extends Arg { // this one is extracted/stamped directly
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withPaths(paths.asJava)
   }
 
-  case class Config(file: Option[Path]) extends Arg {
+  case class Config(file: Option[Path]) extends Arg with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withConfig(jutil.Optional.ofNullable(file.orNull))
   }
 
-  case class ParsedArgs(args: Seq[String]) extends Arg {
+  case class ParsedArgs(args: Seq[String]) extends Arg with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withParsedArguments(args.asJava)
   }
 
-  case class ScalaVersion(version: String) extends Arg {
+  case class ScalaVersion(version: String) extends Arg { //FIXME: with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withScalaVersion(version)
   }
 
-  case class ScalacOptions(options: Seq[String]) extends Arg {
+  case class ScalacOptions(options: Seq[String]) extends Arg { //FIXME: with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withScalacOptions(options.asJava)
   }
 
-  case class Classpath(classpath: Seq[Path]) extends Arg {
+  case class Classpath(classpath: Seq[Path]) extends Arg { //FIXME: with CacheKey {
     override def apply(sa: ScalafixArguments): ScalafixArguments =
       sa.withClasspath(classpath.asJava)
   }
+
+  case object NoCache extends Arg with CacheKey {
+    override def apply(sa: ScalafixArguments): ScalafixArguments =
+      sa // caching is currently implemented in sbt-scalafix itself
```

Comment:
I am assuming `--no-cache` could become a `scalafix-cli` argument in the future if/when caching is implemented upstream.